### DISCUSSION TITLE
Permission For Flows

### DISCUSSION THREAD

**Discussion by imiskolee** [directus/directus#19719] [Feature Requests]
### Summary

allows role can be set flow level permission.

### Motivation

We use direcuts for our operations backend, different teams have different module permissions, we make extensive use of manual trigger flow to build advanced customization, operations colleagues can simply click a button that can complete some complex data operations work, efficient and simple.

However, we found that the current directus can't control the permissions for flow, we hope the operation colleagues can only click the flow button and read the data permissions.

We don't know if we can do this.

Translated with www.DeepL.com/Translator (free version)

### Detailed Design

I would like to add a new Flow column in the Role setting screen, which can be selected in full or in part.

### Must Have:
I would like to add a new Flow column in the Role setting screen, which can be selected in full or in part.

### Drawbacks

N/A

### Alternatives

N/A

### Adoption Strategy

N/A

**Comment by BartTactick** [comment]
It would be nice to have something that dynamicly shows the button and/or gives feedback from the flow itself.
But for now you should be able to check if a user is a certain team/role based on the trigger of the flow (with a condition)

**Reply by imiskolee** [reply]
but the big issue from now is  need admin permission, if not admin permission, the flow button is disabled.

**Reply by AlaaL** [reply]
It works without admin access, first it was not working for me, so i edited the directus_flows permissions to all access for read then return it as it was before to be `trigger equals manual` i think it does not work for old roles but works for new ones ? because i created a new role and it did not have that issue.

image from none admin role
![image](https://github.com/directus/directus/assets/5410074/28801a0b-5ceb-42bb-9e6c-02e067374cbf)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
